Core/clim-core/gadgets/concrete.lisp

```diff
@@ -1583,10 +1583,11 @@ if INVOKE-CALLBACK is given."))
                       ((in-menu (event-sheet event) x y)
                        (handle-event (event-sheet event) event)))))))
         ;; Cleanup and exit
-        (when retain-value
-          (setf (gadget-value parent :invoke-callback t)
-                (gadget-value list-pane)))
-        (disown-frame manager menu-frame)))))
+        (unwind-protect
+             (when retain-value
+               (setf (gadget-value parent :invoke-callback t)
+                     (gadget-value list-pane)))
+          (disown-frame manager menu-frame))))))
```

Comment:
what if the error happens in the handle-event method inside tracking-pointer? shouldn't the whole body after enable-frame be wrapped in the unwind-protect?

perhaps this function could be slightly refactored, i.e instead of popup-init function it could be

```
(with-popup (parent manager frame)
    (list-pane topmost-pane menu-frame)
  (labels …))
```

where with-popup expands to something like
```
(let (,@panes)
  (adopt-frame manager menu-frame)
  (enable-frame menu-frame)
  (unwind-protect (progn ,@body)
    (disown-frame manager menu-frame)))
```